activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -80,12 +80,34 @@ def without(*elements)
   #   [{ id: 1, name: "David" }, { id: 2, name: "Rafael" }].pluck(:id, :name)
   #     => [[1, "David"], [2, "Rafael"]]
   def pluck(*keys)
-    if keys.many?
-      map { |element| keys.map { |key| element[key] } }
-    else
-      map { |element| element[keys.first] }
+    entries = map { |element| keys.map { |key| pluck_single(element, key) } }
+
+    return entries if keys.many?
+
+    entries.flatten!
+  end
+
+  # Queries an object (in particular, an ActiveRecord model, Hash, or Struct)
+  # for a particular value using #[], #fetch, or #send, as appropriate,
+  # returning nil if the value is not found.
+  #
+  #   pluck_single({ name: "David" }, :name)
+  #     => "David"
+  #
+  #   pluck_single({ name: "David" }, :email)
+  #     => nil
+  def pluck_single(element, key)
```

Comment:
I mean, we need to add `# :nodoc:` to this method because it is private API and should not be documented :smile: